### PR TITLE
Fix JstPipelineTest

### PR DESCRIPTION
**Pull request by marcushoepfner**
Issue: https://github.com/eclipse-platform/eclipse.platform.ui/issues/665

### CODE REVIEW

File: tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/jst/JstPipelineTest.java [hunk 1]
```diff
@@ -43,50 +47,43 @@ public JstPipelineTest() {
...
-				TEST_CONTENT_JST }, false);
+		_contentService.bindExtensions(
+				new String[] { COMMON_NAVIGATOR_RESOURCE_EXT, COMMON_NAVIGATOR_JAVA_EXT, TEST_CONTENT_JST }, false);
 
 		// Note this test will fail showing only one if the JDT stuff
 		// is not included in the executing bundles (which it normally is)
-		assertEquals(
-				"There should be two visible extensions for the pipeline viewer.",
-				3, _contentService.getVisibleExtensionIds().length);
+		assertEquals("There should be two visible extensions for the pipeline viewer.", 3,
```
Comment (vogella) [inline tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/jst/JstPipelineTest.java]: I know this is old, but "two visible extension" vs using 3 sounds something we should adjust.

**Review by vogella — CHANGES_REQUESTED** [review]
Thanks for working on this Marcus.

File: tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/jst/JstPipelineTest.java (hunk 1 above)
Comment (marcushoepfner) [reply tests/org.eclipse.ui.tests.navigator/src/org/eclipse/ui/tests/navigator/jst/JstPipelineTest.java]: Done. Did not see it.

**Comment by marcushoepfner** [comment]
> Thanks for working on this Marcus.

Sure. Don't really understand the test. But I can see, that the Test is not working on Jenkins (mac). While locally it works on Mac (in Eclipse as PDE test).

Let's wait for the build to finish and check the console.

**Comment by marcushoepfner** [comment]
> Thanks for working on this Marcus.

Sure. But I got some errors I don't understand. In order to understand the problem I added some system outs.

On Linux or Windows on Jenkins (both are similar), the children of the "Compressed Java" node are like this:
`Starting here ... ==================================================
Compressed Libraries
java.base - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.compiler - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.datatransfer - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.desktop - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.instrument - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.logging - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.management - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.management.rmi - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.naming - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.net.http - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.prefs - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.rmi - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.scripting - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.se - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.security.jgss - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.security.sasl - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.sql - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.sql.rowset - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.transaction.xa - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.xml - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
java.xml.crypto - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.accessibility - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.attach - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.compiler - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.dynalink - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.httpserver - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.incubator.foreign - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.incubator.vector - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.internal.ed - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.internal.jvmstat - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.internal.le - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.internal.opt - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jartool - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.javadoc - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jconsole - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jdi - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jdwp.agent - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jfr - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jshell - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.jsobject - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.management - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.management.agent - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.management.jfr - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.net - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.nio.mapmode - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.sctp - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.security.auth - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.security.jgss - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.unsupported - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.unsupported.desktop - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
jdk.xml.dom - /usr/lib/jvm/temurin-17-jdk-amd64/lib/jrt-fs.jar
src
Done======================`

While on Mac (Jenkins) like this:
`Starting here ... ==================================================
(default package)
a.b.c.d
Compressed Libraries
Done======================`

Locally on my mac machine it looks on Jenkins for Linux/Windows.

Is the JDT/JVM setup wrongly for Mac on Jenkins?

I have no clue how to fix this.
@vogella  could you assist or do you know someone who could help?

**Comment by vogella** [comment]
I have no idea about Mac in general.

**Comment by marcushoepfner** [comment]
who set up the jenkins (for mac)?

**Comment by laeubi** [comment]
> who set up the jenkins (for mac)?

There is no mac Jenkins instance for platform, we have only mac runners on github.
You can see the full setup here:
https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/.github/workflows/mavenBuild.yml

But mac test sometime require special care if they using `jdt.launching`, not sure if it might be the case here:
- https://github.com/eclipse-pde/eclipse.pde/pull/545

**Comment by marcushoepfner** [comment]
> > who set up the jenkins (for mac)?
> 
> There is no mac Jenkins instance for platform, we have only mac runners on github. You can see the full setup here: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/.github/workflows/mavenBuild.yml
> 
> But mac test sometime require special care if they using `jdt.launching`, not sure if it might be the case here:
> 
>     * [Use p2.inf to pull in depedency to org.eclipse.jdt.launching.macosx eclipse-pde/eclipse.pde#545](https://github.com/eclipse-pde/eclipse.pde/pull/545)

will try p2.inf file

**Comment by marcushoepfner** [comment]
> > > who set up the jenkins (for mac)?
> > 
> > 
> > There is no mac Jenkins instance for platform, we have only mac runners on github. You can see the full setup here: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/.github/workflows/mavenBuild.yml
> > But mac test sometime require special care if they using `jdt.launching`, not sure if it might be the case here:
> > ```
> > * [Use p2.inf to pull in depedency to org.eclipse.jdt.launching.macosx eclipse-pde/eclipse.pde#545](https://github.com/eclipse-pde/eclipse.pde/pull/545)
> > ```
> 
> will try p2.inf file

Did work! Thanks for the hint @laeubi

**Comment by marcushoepfner** [comment]
@vogella Seems that I need a re-review from you side. Only then I can merge. Thanks

**Review by vogella — APPROVED** [review]
The test failure in org.eclipse.ui.tests.navigator looks unrelated

**Pull request merged**